test_util/helpers.py

```diff
@@ -189,3 +248,15 @@ def remove_file():
     # Attempt to remove the file upon normal interpreter exit.
     atexit.register(remove_file)
     return temp_path
+
+
+def lazy_property(property_fn):
+    cache_name = '{}_cached'.format(property_fn.__name__)
```

Comment:
`__<name>_cached` so that we make this a private member plese using python's underscore public / private interface standard